Objects/unicodeobject.c

```diff
@@ -7314,14 +7308,10 @@ decode_code_page_errors(UINT code_page,
         }
     }
 
-    /* write a NUL character at the end */
-    *out = 0;
-
-    /* Extend unicode object */
+    /* Shrink the buffer */
     outsize = out - startout;
-    assert(outsize <= PyUnicode_WSTR_LENGTH(*v));
-    if (unicode_resize(v, outsize) < 0)
-        goto error;
+    assert(outsize <= *bufsize);
+    *bufsize = outsize;
```

Comment:
Example of place where it becomes unclear if bufsize is the allocation size or the current position :-/